lib/ansible/modules/cloud/vmware/vmware_local_user_manager.py

```diff
@@ -111,71 +132,240 @@ def process_state(self):
             self.module.fail_json(msg=str(e))
 
     def check_local_user_manager_state(self):
+        """Check user account"""
         user_account = self.find_user_account()
-        if not user_account:
-            return 'absent'
-        else:
+        if user_account:
+            self.user_account = user_account[0]
             return 'present'
+        return 'absent'
 
     def find_user_account(self):
-        searchStr = self.local_user_name
-        exactMatch = True
-        findUsers = True
-        findGroups = False
-        user_account = self.content.userDirectory.RetrieveUserGroups(None, searchStr, None, None, exactMatch, findUsers, findGroups)
+        """Try to find the user account"""
+        search_string = self.local_user_name
+        exact_match = True
+        find_users = True
+        find_groups = False
+        user_account = self.content.userDirectory.RetrieveUserGroups(
+            None, search_string, None, None, exact_match, find_users, find_groups
+        )
         return user_account
 
     def create_account_spec(self):
+        """Create account specification"""
         account_spec = vim.host.LocalAccountManager.AccountSpecification()
         account_spec.id = self.local_user_name
         account_spec.password = self.local_user_password
         account_spec.description = self.local_user_description
         return account_spec
 
-    def state_create_user(self):
-        account_spec = self.create_account_spec()
-
+    def set_permissions(self, user, role_id):
+        """Set specified permissions on an entity"""
+        permission_spec = vim.AuthorizationManager.Permission()
+        permission_spec.principal = user
+        permission_spec.group = False
+        permission_spec.roleId = role_id
+        permission_spec.propagate = True
```

Comment:
Can we make this parameter configurable just like web ui ?

Reply:
sure, but you can only configure the permissions on the root folder (entity).